lib/hooks/useDragNDrop.js

```diff
@@ -0,0 +1,58 @@
+import { useState } from 'react';
+import classNames from 'classnames';
+
+const useDragAndDrop = () => {
+  const [highlight, setHighlight] = useState(false);
+
+  const fileListToArray = list => {
+    const array = [];
+
+    for (let i = 0; i < list.length; i += 1) {
+      array.push(list.item(i));
+    }
+    return array;
+  };
+
+  const onDragOver = evt => {
+    evt.preventDefault();
+    setHighlight(true);
+  };
+
+  const onDragLeave = e => {
+    e.preventDefault();
+    setHighlight(false);
+  };
+
+  const onDrop = event => {
+    event.preventDefault();
+    const { files } = event.dataTransfer;
+    const array = fileListToArray(files);
+    return array;
+  };
+
+  const onFilesAdded = evt => {
+    const { files } = evt.target;
+    const array = fileListToArray(files);
+    return array;
+  };
+
+  const getHighlightClass = className => {
+    const styleCls = classNames(className, {
+      highlight,
+    });
+    return styleCls;
+  };
+
+  return {
```

Comment:
@salilkapoor is there a way someone could filter out files being uploaded through drag and drop and selection? I think it input[type="file"] its easy to set file types but drag and drop would need some mechanics for it. 

Like if the drop only supports .png then other files should be excluded with a message to user. 

Also there should be a way to set upload file-size limit, single or collective.

Sorry for not defining it as par of  requirement.